src/cascadia/TerminalApp/FilteredCommand.cpp

```diff
@@ -61,49 +60,40 @@ namespace winrt::TerminalApp::implementation
         }
     }
 
-    void FilteredCommand::_update()
+    static std::tuple<std::vector<winrt::TerminalApp::HighlightedRun>, int32_t> _matchedSegmentsAndWeight(const std::shared_ptr<fzf::matcher::Pattern>& pattern, const winrt::hstring& haystack)
     {
-        std::vector<winrt::TerminalApp::HighlightedTextSegment> segments;
-        const auto commandName = _Item.Name();
+        std::vector<winrt::TerminalApp::HighlightedRun> segments;
         int32_t weight = 0;
 
-        if (!_pattern || _pattern->terms.empty())
+        if (pattern && !pattern->terms.empty())
         {
-            segments.emplace_back(winrt::TerminalApp::HighlightedTextSegment(commandName, false));
+            if (auto match = fzf::matcher::Match(haystack, *pattern.get()); match)
+            {
+                auto& matchResult = *match;
+                weight = matchResult.Score;
+                segments.resize(matchResult.Runs.size());
+                std::transform(matchResult.Runs.begin(), matchResult.Runs.end(), segments.begin(), [](auto&& run) -> winrt::TerminalApp::HighlightedRun {
+                    return { run.Start, run.End };
+                });
+            }
         }
-        else if (auto match = fzf::matcher::Match(commandName, *_pattern.get()); !match)
+        return { std::move(segments), weight };
+    }
+
+    void FilteredCommand::_update()
+    {
+        auto itemName = _Item.Name();
```

Comment:
```suggestion
```
unused variable

Reply:
ah, that belongs to a different PR. thanks ;P